impl/src/main/java/jakarta/faces/convert/UUIDConverter.java

```diff
@@ -0,0 +1,111 @@
+     * <p>
+     * The standard converter id for this converter.
+     * </p>
+     */
+    public static final String CONVERTER_ID = "jakarta.faces.UUID";
+
+    /**
+     * <p>
+     * The message identifier of the {@link jakarta.faces.application.FacesMessage} to be created if the conversion to <code>UUID</code> fails.
+     * The message format string for this message may optionally include the following placeholders:
+     * <ul>
+     * <li><code>{0}</code> replaced by the unconverted value.</li>
+     * <li><code>{1}</code> replaced by an example value.</li>
+     * <li><code>{2}</code> replaced by a <code>String</code> whose value is the label of the input component that produced this message.</li>
+     * </ul>
+     */
+    public static final String UUID_ID = "jakarta.faces.converter.UUIDConverter.UUID";
+
+    /**
+     * <p>
+     * The message identifier of the {@link jakarta.faces.application.FacesMessage} to be created if the conversion of the <code>UUID</code> value to <code>String</code> fails.
+     * The message format string for this message may optionally include the following placeholders:
+     * <ul>
+     * <li><code>{0}</code> relaced by the unconverted value.</li>
+     * <li><code>{1}</code> replaced by a <code>String</code> whose value is the label of the input component that produced this message.</li>
+     * </ul>
+     */
+    public static final String STRING_ID = "jakarta.faces.converter.STRING";
+
+    // ------------------------------------------------------- Converter Methods
+
+    /**
+     * @throws ConverterException {@inheritDoc}
+     * @throws NullPointerException {@inheritDoc}
+     */
+    @Override
+    public UUID getAsObject(FacesContext context, UIComponent component, String value) {
+
+        if (context == null || component == null) {
+            throw new NullPointerException();
+        }
+
+        if (value == null || value.isBlank()) {
+            return null;
+        }
+
+        try {
+            return UUID.fromString(value);
+        } catch (IllegalArgumentException e) {
+            throw new ConverterException(MessageFactory.getMessage(context, UUID_ID, value, "29573f55-4254-4afa-9ca6-6b5ae6c7ab6e", MessageFactory.getLabel(context, component)), e);
+        } catch (Exception e) {
+            throw new ConverterException(e);
+        }
+    }
+
+    /**
+     * @throws ConverterException {@inheritDoc}
+     * @throws NullPointerException {@inheritDoc}
+     */
+    @Override
+    public String getAsString(FacesContext context, UIComponent component, UUID value) {
+
+        if (context == null || component == null) {
+            throw new NullPointerException();
+        }
+
+        if (value == null) {
+            return "";
+        }
+
+        try {
+            return value.toString();
+        } catch (Exception e) {
+            throw new ConverterException(MessageFactory.getMessage(context, STRING_ID, value, MessageFactory.getLabel(context, component)), e);
```

Comment:
Just out of curiosity - assuming `value.toString()` throws - what is expected to happen with the message [for `ConverterException`] preparation? I suppose another `value.toString` call to be made...

Reply:
Sharp. This logic was just extracted as a template from existing faces converters.